platform/drm/cog-drm-renderer.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * cog-drm-renderer.c
+ * Copyright (C) 2021 Igalia S.L.
+ *
+ * Distributed under terms of the MIT license.
+ */
+
+#include "cog-drm-renderer.h"
+
+void
+cog_drm_renderer_destroy(CogDrmRenderer *self)
+{
+    if (self) {
+        if (self->destroy)
+            self->destroy(self);
+        else
+            g_free(self);
```

Comment:
While it may make sense if I knew the codebase better, this looks like weird behaviour to me... asserting self->destroy and calling it feels like the more natural thing to me, surely every renderer should have a destructor?

Reply:
So far both renderers added by this patch set have one, so let's `g_assert(self->destroy)` for the sake of simplicity. If the need to have a renderer without a destruction function comes up later, we can always rethink this.

Reply:
Also I think it's somewhat silly to have a `cog-drm-renderer.c` file with a single two-line function so I will remove it and put the `cog_drm_renderer_destroy()` in the header as a `static inline`.